avo-hq/avo
# Feature/better error for missing resource

# Description

Fixes [# 1960](https://github.com/avo-hq/avo/issues/1960)

We're adding a better handling for errors when we encounter a missing resource on association fields.

Right now the error is unhelpful and requires developer to understand the internal logic of Avo. 
We want to give a better Dev Experience and point to solutions.

# Checklist:


## Screenshots

Error:

UPDATED WITH NEW MESSAGE
<img width="1179" alt="image" src="https://github.com/avo-hq/avo/assets/12682792/a35de12a-cfe9-45df-bcd0-c709aa07d8d5">


## Video

BEFORE APPLYING REVIEW SUGGESTIONS
https://github.com/avo-hq/avo/assets/12682792/3b4f14f8-700a-4770-ba37-cecd66683052


## Manual review steps

1. Go to the dummy app to Teams page and choose any Team, add `show_location_field=1` query param
1. On the Team show page go to the bottom and observe what happens with the slot for `locations` field.
1. Go to the Courses page and choose any Course, add `show_location_field=1` query param
1. On the Course show page go to the bottom and observe what happens with the slot for `locations` field.
1. Go to the Stores page , add `show_location_field=1` query param and see the error
1. Go to the Events page, add `show_location_field=1` query param and see the error

Manual reviewer: please leave a comment with output from the test if that's the case.
